File: EdgeData.py


class EdgeData:

    def __init__(self, src, dest, weight, info=None, tag=None):
        """
        constructor for object EdgeData
        """
        self.src = src
        self.dest = dest
        self.weight = weight
        self.info = info
        self.tag = tag

    def getSrc(self):
        """
        return the id of the source node of this edge
        """
        return self.src

    def getDest(self):
        """
        return the id of the destination node of this edge
        """
        return self.dest

    def getWeight(self):
        """
        return the weight of this edge (positive value).
        """
        return self.weight

    def getInfo(self):
        """
        return the remark (meta data) associated with this edge
        """
        return self.info

    def getTag(self):
        """
        which can be used in algorithms
        temporal data (aka color: white, gray, black)
        """
        return self.tag

    def setInfo(self, info):
        """
        Allows changing the remark (meta data) associated with this edge.
        """
        self.info = info

    def setTag(self, tag):
        """
        this method allows setting the tag value for temporal marking an edge -
        common practice for marking by algorithms.
        """
        self.tag = tag

File: java/EdgeData.java
package ex2;

public class EdgeData implements edge_data {
	private int src;
	private int dest;
	private double weight;
	private String info;
	private int tag;
	
	public EdgeData(int src,int dest, double weight) {	
		this.src=src;
		this.dest=dest;
		this.weight=weight;
	}
		
		/**
	 * The id of the source node of this edge.
	 * @return
	 */
	public int getSrc() {
		return this.src;
	}
	/**
	 * The id of the destination node of this edge
	 * @return
	 */
	public int getDest() {
		return this.dest;
	}
	/**
	 * @return the weight of this edge (positive value).
	 */
	public double getWeight() {
		return this.weight;
	}
	/**
	 * Returns the remark (meta data) associated with this edge.
	 * @return
	 */
	public String getInfo() {
		return this.info;
	}
	/**
	 * Allows changing the remark (meta data) associated with this edge.
	 * @param s
	 */
	public void setInfo(String s) {
		this.info=s;
	}
	/**
	 * Temporal data (aka color: e,g, white, gray, black) 
	 * which can be used be algorithms 
	 * @return
	 */
	public int getTag() {
		return this.tag;
	}
	/** 
	 * This method allows setting the "tag" value for temporal marking an edge - common
	 * practice for marking by algorithms.
	 * @param t - the new value of the tag
	 */
	public void setTag(int t) {
		this.tag=t;
	}
}


